buildutils/torque.spec.in

```diff
@@ -61,9 +61,9 @@
 %define ac_trqauthd_sock_dir %{?trqauthd_sock_dir:%{trqauthd_sock_dir}}%{!?trqauthd_sock_dir:/tmp}
 
 ### Build Requirements
-%define breq_cpuset   %{?with_cpuset:hwloc-devel}
+%define breq_cpuset   %{!?hwloc_path:%{?with_cpuset:hwloc-devel}}
 %define breq_gui      %{?with_gui:tcl-devel tk-devel tclx}
-%define breq_numa     %{?with_cpuset:hwloc-devel}
+%define breq_numa     %{!?hwloc_path:%{?with_cpuset:hwloc-devel}}
 %define breq_munge    %{?with_munge:munge-devel}
```

Comment:
First, to clarify, the `configure.ac` and other spec file changes look _fine_.  My concerns are _only_ WRT disabling the `hwloc-devel` buildreq when a path to `hwloc` is supplied.

As a general rule, if a person is going through the effort of turning their TORQUE build into an RPM (which is what `make rpm` does, and that's the only scenario in which `RPM_AC_OPTS` comes into play), it's because they either have a policy to build all system-level software packages for their systems into RPMs (which is definitely my strong recommendation and preference, not to mention an industry best practice), or because they at least see the benefit and wisdom of having one's system-level packages handled by a centralized package management system.

As such, it doesn't make much sense to go to the trouble of building TORQUE as an RPM but not build an RPM for hwloc.  And, in fact, having the upstream spec file for TORQUE support mixing-and-matching of packages -- where some things are installed via RPM and others not, possibly in varying/unpredictable paths -- would have significant costs in both readability and supportability.  So the current spec file only supports building TORQUE RPMs from dependencies which must also be installed as RPMs.  To do otherwise would quickly become unmaintainable, IMHO.

Those wishing/needing to build RPMs for TORQUE without building RPMs for one or more of the prerequisite packages can always patch their local spec file, or simply use `rpmbuild --nodeps` to build.  But IMHO, supplying `--with-hwloc-path=/usr/local` does not remove the need for hwloc at build time -- it simply indicates where to find it -- and so the spec file should not remove the buildreq either.

This is especially true for packaging infrastructures like Fedora and OBS which build all packages from clean chroot jails.  ALL dependencies must be accurately reflected in the spec file, or the package simply won't build correctly!  So removing or disabling a buildreq is a pretty big deal.  :-)

> Furthermore, if the `BuildRequires` is necessary, then maybe the version should be specified in the specfile too (like in configure.ac)

If `configure.ac` specifies a particular minimum version requirement for `hwloc` and/or `hwloc-devel`, then yes, the spec file should too.

Does that help at all?